src/hooks/click-outside.js

```diff
@@ -0,0 +1,25 @@
+import { useEffect } from 'react'
+
+export function useClickOutside(el, onClick) {
+  const handler = (ev) => {
+    let target = ev.target
+
+    if (target === el.current) {
```

Comment:
I'd maybe be a little less restrictive on the `el` parameter here.

An array of nodes might be more powerful. Accepting either single or an array. I'd then make the checks a little more thorough. Making use of `.contains` would be ideal 👍 


This isn't optimal but something along these lines potentially. The reason for this being you might have a layout component whose children are the main content of the app. An outside click would never fire in this case. If you can pass an `Array` of refs, you can avoid this issue.
```
const handler = (ev) => {
  const target = ev.target
  let outside = false
  if (el.current.length) {
    for (const node of el.current) {
      if (!node.contains(target)) outside = true
    }
  } else if (!el.contains(target)) {
    outside = true
  }
  if (outside) onClick(ev)
}
```

Reply:
> you might have a layout component whose children are the main content of the app. An outside click would never fire in this case.

Just trying to understand this use case... what is `ev.target` in this scenario, and what is `el`?

Reply:
For example, we have projects where the router instance is wrapped in a layout component that compromises of the app content. You'd likely use an "off" click here for a burger menu style component. But everything on the page is a child of the element. And `parentElement` won't detect all scenarios. It's purely to be more flexible as people will build things in all different ways.

One example could be that you have a header and side menu. When it's open, you want to close it on "off" click. But you don't want it to close when you click the header or side menu or any descendant of each.

That way I could have something like `useClickOutside([header.current, sidemenu.current], handleClickOutside)`

Reply:
Thanks, this makes sense. We solve this issue in a current project by having our `ClickElsewhere` component take an `ignore' array of class-names. We also ignore anything that has a `data-click-elsewhere-ignore` attribute. I was going to follow up with another PR that added something like this, but I think your suggestion is better.

Reply:
@jh3y does https://github.com/nearform/react-browser-hooks/pull/100/commits/a565ab27c0aa1eb144c44f97ec35e06c2a26ac86 improve matters for you?

This solves the case where we want to not call `onClick` when clicking on certain components.

Your earlier comments seem to also suggest another case where `onClick` is _not_ called where we _do_ want it to be, if so I'm still not sure I understand this case (sorry).